space/internal/components/aclobjectmanager/aclobjectmanager.go

```diff
@@ -157,14 +157,19 @@ func (a *aclObjectManager) process() {
 
 func (a *aclObjectManager) processAcl() (err error) {
 	var (
-		common   = a.sp.CommonSpace()
-		acl      = common.Acl()
-		aclState = acl.AclState()
+		common         = a.sp.CommonSpace()
```

Comment:
Tbh looking at this now, I would've done it like this:
- remove checking for self status from process states, to make it simpler
- create another method `isAccountRemoved` which would be very simple: call `aclState.Permissions(aclState.Identity())`check if the account was removed (you can also add there the logic with upToDate)